### PR TITLE
DC-126: Deduplicate the metric on the query results

### PR DESCRIPTION
**Pull request by patrick-schweizer**
_TimeSeriesStorage.getTimeseries(TimeSeriesFetchRequest request)_ returns a List of Samples.
Each Sample contains the same Metric Object.
This PR deduplicates that and returns only one Metric per getTimeseries request.

Jira: https://issues.opennms.org/browse/DC-126

### CODE REVIEW

File: api/src/main/java/org/opennms/integration/api/v1/timeseries/TimeSeriesStorage.java [hunk 1]
```diff
@@ -51,9 +54,106 @@ public interface TimeSeriesStorage {
...
-    /** Returns the data for the given metrics for the given time period. */
+    /** Returns the data for the given metrics for the given time period.
+     *  Deprecated: use getTimeSeriesData() instead.
+     *  This method will only be called if getTimeSeriesData is not implemented yet.
+     */
+    @Deprecated
     List<Sample> getTimeseries(TimeSeriesFetchRequest request) throws StorageException;
 
+    /** Returns the data for the given metric for the given time period. */
+    default TimeSeriesData getTimeSeriesData(TimeSeriesFetchRequest request) throws StorageException {
```
Comment (patrick-schweizer) [inline api/src/main/java/org/opennms/integration/api/v1/timeseries/TimeSeriesStorage.java:65]: to be discussed: Do we want to provide the default implementation for backwards compatibility?

Comment (j-white) [reply api/src/main/java/org/opennms/integration/api/v1/timeseries/TimeSeriesStorage.java:65]: yes, we should remain backwards compatible

**Comment by j-white** [comment]
Can we target the `release-1.x` branch with this change? Could be part of a `1.1.0`.

File: api/src/main/java/org/opennms/integration/api/v1/timeseries/TimeSeriesStorage.java (hunk 1 above)
Comment (patrick-schweizer) [reply api/src/main/java/org/opennms/integration/api/v1/timeseries/TimeSeriesStorage.java:65]: ok

**Comment by patrick-schweizer** [comment]
> Can we target the `release-1.x` branch with this change? Could be part of a `1.1.0`.

done.

**Review by j-white — APPROVED** [review]
LGTM

**Pull request merged**